contrib/opentimelineio_contrib/adapters/xges.py

```diff
@@ -190,507 +300,925 @@ def to_rational_time(self, ns_timestamp):
-                    self._create_otio_gap(
-                        self.to_rational_time(0),
-                        (clip_offset - track.duration())
-                    )
-                )
-
-            track.append(otio_clip)
-
-        return track
-
-    def _get_clip_name(self, clip, all_names):
-        i = 0
-        tmpname = name = clip.get("name", GstStructure(
-                                  clip.get("properties", "properties;")).get("name"))
-        while True:
-            if tmpname not in all_names:
-                all_names.add(tmpname)
-                return tmpname
-
-            i += 1
-            tmpname = name + '_%d' % i
+            clip_type = self._get_attrib(clip, "type-name", str)
+            start = self._get_attrib(clip, "start", int)
+            inpoint = self._get_attrib(clip, "inpoint", int)
+            duration = self._get_attrib(clip, "duration", int)
+            otio_composable = None
+            if clip_type == "GESTransitionClip":
+                otio_composable = self._otio_transition_from_clip(clip)
+            elif clip_type == "GESUriClip":
+                otio_composable = self._otio_item_from_uri_clip(clip)
+            else:
+                # TODO: support other clip types
+                # maybe represent a GESTitleClip as a gap, with the text
+                # in the metadata?
+                # or as a clip with a MissingReference?
+                show_ignore(
+                    "Could not represent %s clip type" % (clip_type))
+
+            if isinstance(otio_composable, otio.schema.Transition):
+                otio_transitions.append({
+                    "transition": otio_composable,
+                    "start": start, "duration": duration})
+            elif isinstance(otio_composable, otio.core.Item):
+                otio_items.append({
+                    "item": otio_composable, "start": start,
+                    "inpoint": inpoint, "duration": duration})
+        return otio_items, otio_transitions
 
-    def _create_otio_transition(self, clip, all_names):
-        start = self.to_rational_time(int(clip.attrib["start"]))
-        end = start + self.to_rational_time(int(clip.attrib["duration"]))
-        cut_point = otio.opentime.RationalTime(
-            (end.value - start.value) / 2.0,
-            start.rate
-        )
+    @staticmethod
+    def _item_gap(second, first):
+        if second is None:
+            return 0
+        if first is None:
+            return second["start"]
+        return second["start"] - first["start"] - first["duration"]
 
-        return otio.schema.Transition(
-            name=self._get_clip_name(clip, all_names),
+    def _add_otio_composables_to_track(self, track, items, transitions):
+        """
+        Insert items and transitions into the track with correct
+        timings.
+        items argument should be an array of dicts, containing an otio
+        item, and its start, inpoint and duration times (from the
```

Comment:
you might want to specify the coordinate space here that you're expecting - ie "its start, input and duration time in the clip's coordinate space" vs "its start, input and duration time in the track's coordinate space".

Reply:
Sorry, this is an overlap of the language used in the `GES` library and the `otio` library. These values (`start`, `duration`, `inpoint`) should correspond to the xges attributes of the same name. I suppose these would relative to the track's coordinate space.